core/src/config.rs

```diff
@@ -22,3 +24,19 @@ pub enum Letterbox {
     #[serde(rename = "on")]
     On,
 }
+
+impl FromStr for Letterbox {
+    type Err = &'static str;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        if s == "off" {
+            Ok(Letterbox::Off)
+        } else if s == "fullscreen" {
+            Ok(Letterbox::Fullscreen)
+        } else if s == "on" {
+            Ok(Letterbox::On)
+        } else {
+            Err("Not a valid letterbox value")
+        }
```

Comment:
Let's do a match statement here
```suggestion
        match s {
            "off" => Ok(Letterbox::Off),
            "fullscreen" => Ok(Letterbox::Fullscreen),
            "on" => Ok(Letterbox::On),
            _ => Err("Not a valid letterbox value")
        }
```